Nodejs/Product/Npm/SPI/RootPackage.cs

```diff
@@ -25,13 +25,34 @@ internal class RootPackage : IRootPackage {
             string fullPathToRootDirectory,
             bool showMissingDevOptionalSubPackages,
             Dictionary<string, ModuleInfo> allModules = null,
-            int depth = 0) {
+            int depth = 0)
+        :
+            this(LoadRootPackageJson(fullPathToRootDirectory), fullPathToRootDirectory, showMissingDevOptionalSubPackages, allModules, depth)
+        { }
+    
+        public RootPackage(
+            IPackageJson packageJson,
+            string fullPathToRootDirectory,
+            bool showMissingDevOptionalSubPackages,
+            Dictionary<string, ModuleInfo> allModules = null,
+            int depth = 0)
```

Comment:
Why the extra constructor if it's only being used once? Again, just makes things less readable because you have to follow a flow chain when reading through the code.